display_information/update-graphics/src/main/java/com/esri/samples/update_graphics/UpdateGraphicsSample.java

```diff
@@ -167,39 +159,68 @@ public void start(Stage stage) {
 
       mapView.setOnMouseClicked(e -> {
         if (e.getButton() == MouseButton.PRIMARY && e.isStillSincePress()) {
+
           // clear any selected graphic
           graphicsOverlay.clearSelection();
 
+          // set the cursor to default
+          mapView.setCursor(Cursor.DEFAULT);
+
           // create a point from location clicked
-          Point2D mapViewPoint = new Point2D(e.getX(), e.getY());
-
-          if (isUpdateLocationActive) {
-            // add new location to selected graphic
-            Point mapPoint = mapView.screenToLocation(mapViewPoint);
-            selectedGraphic.setGeometry(mapPoint);
-            isUpdateLocationActive = false;
-          } else {
-            // identify the graphic that was selected
-            identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
-
-            identifyGraphics.addDoneListener(() -> {
-              try {
-                if (!identifyGraphics.get().getGraphics().isEmpty()) {
-                  // store the selected graphic
-                  selectedGraphic = identifyGraphics.get().getGraphics().get(0);
-                  selectedGraphic.setSelected(true);
-                  String style = ((SimpleMarkerSymbol) selectedGraphic.getSymbol()).getStyle().toString();
-                  symbolBox.getSelectionModel().select(style);
-
-                  enableUI(false);
-                } else {
-                  enableUI(true);
-                }
-              } catch (Exception x) {
-                new Alert(Alert.AlertType.ERROR, "Error identifying clicked graphic").show();
+          mapViewPoint = new Point2D(e.getX(), e.getY());
+
+          // identify graphics on the graphics overlay
+          identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
+
+          identifyGraphics.addDoneListener(() -> {
+            try {
+              if (!identifyGraphics.get().getGraphics().isEmpty()) {
+                // get the first identified graphic
+                selectedGraphic = identifyGraphics.get().getGraphics().get(0);
+                // select the identified graphic
+                selectedGraphic.setSelected(true);
+                // update the drop down box with the identified graphic's current symbol
+                String style = ((SimpleMarkerSymbol) selectedGraphic.getSymbol()).getStyle().toString();
+                symbolBox.getSelectionModel().select(style);
+                // show the UI
+                disableUI(false);
+              } else {
+                disableUI(true);
               }
-            });
-          }
+            } catch (Exception x) {
+              new Alert(Alert.AlertType.ERROR, "Error identifying clicked graphic").show();
+            }
+          });
+        }
+      });
+
+      mapView.setOnMouseDragged(e -> {
+        if (selectedGraphic.isSelected()) {
```

Comment:
It looks like we need a null check here, to avoid a `NullPointerException` when no graphics is selected and we're dragging around:
```suggestion
        if (selectedGraphic != null && selectedGraphic.isSelected()) {
```